lib/posix/options/pthread.c

```diff
@@ -599,6 +638,20 @@ int pthread_create(pthread_t *th, const pthread_attr_t *_attr, void *(*threadrou
 		t->attr = *(struct posix_thread_attr *)_attr;
 	}
 
+	struct posix_thread *self = NULL;
+
+	K_SPINLOCK(&pthread_pool_lock) {
+		self = to_posix_thread(pthread_self());
+		if (self == NULL) {
+			K_SPINLOCK_BREAK;
+		}
+		if (self->attr.inheritsched == PTHREAD_INHERIT_SCHED) {
+			t->attr.priority = self->attr.priority;
+			t->attr.schedpolicy = self->attr.schedpolicy;
+			t->attr.inheritsched = self->attr.inheritsched;
+		}
```

Comment:
Something like this could probably help, but it would be better to move it into the block ending on line 639.

```suggestion
		if (t->attr.inheritsched == PTHREAD_INHERIT_SCHED) {
     struct k_thread *self = k_current_get();

			/* There might be a function named posix_to_zephyr_priority() ? */
			t->attr.priority /* = ...(self) */;
			t->attr.schedpolicy = /* = ...(self) */;
		}
```